frontend/packages/ux-editor/src/utils/generateId.ts

```diff
@@ -0,0 +1,22 @@
+import { IFormLayout } from '../types/global';
+import { generateRandomId } from 'app-shared/utils/generateRandomId';
+
+export const generateTextResourceId = (
+  layoutName: string,
+  componentId: string,
+  textKey: string
+) => {
+  return `${layoutName}-${componentId}-${textKey}`;
```

Comment:
Should these strings be passed through a normalization step? Like a slugify function or similar?

Reply:
We could do that, although we have validation on each of them separately to ensure there are no whitespaces f.ex., so I'm not sure we actually need to.